app/src/main/java/me/guillaumin/android/osmtracker/db/model/TrackStatistics.java

```diff
@@ -0,0 +1,246 @@
+package me.guillaumin.android.osmtracker.db.model;
+
+import me.guillaumin.android.osmtracker.db.TrackContentProvider;
+import me.guillaumin.android.osmtracker.db.TrackContentProvider.Schema;
+import android.content.ContentResolver;
+import android.database.Cursor;
+import android.location.Location;
+import android.util.Log;
+
+/**
+ * Represents statistics, such as  total length and maximum speed, for a Track
+ *
+ * @author Arseniy Lartsev
+ *
+ */
+public class TrackStatistics {
+	private long trackId;
+	private long pointCount;
+	private float length;
+	private float maxSpeed;
+	private double lastLatitude;
+	private double lastLongitude;
+	private long lastTime;
+	private float timeMoving;
+
+	/**
+	 * Copypasta from osmand
```

Comment:
I think I'll ditch this code instead ☺ Turns out treating the Earth as a sphere, which only introduces 0.5% error, brings about a factor of 2 speedup, which means it's a better deal (the timings I've reported have been measured with this code replaced with the trivial sphere formula).